### PR TITLE
Get the python code to snitch it's version

### PR DESCRIPTION
**Pull request by foursixnine**
Ensure there's evidence of the Python version that is embedded in the perl extension

While looking at Inline-Perl with @josegomezr, we realized that there's no information about the python version the extension is compiled with

### CODE REVIEW

File: t/08-autotest.t [hunk 1]
```diff
@@ -316,9 +316,13 @@ subtest 'load test successfully when CASEDIR is a relative path' => sub {
...
 subtest python => sub {
-    stderr_like {
+    my $python_module_regex = qr{scheduling pythontest tests/pythontest};
+    my $python_version_regex = qr{Using python version};
+
+    combined_like {
         lives_ok { autotest::loadtest('tests/pythontest.py') } 'can load test module'
-    } qr{scheduling pythontest tests/pythontest.py}, 'python pythontest module referenced';
+    } qr{$python_module_regex+$python_version_regex}, 'python pythontest module referenced';
+
```
Comment (josegomezr) [inline t/08-autotest.t:325]: why not running `autotest::loadtest('tests/pythontest.py')` for each assertion?

one for testing that `scheduling pythontest ...` exists. one for the python version.

File: autotest.pm [hunk 2]
```diff
@@ -126,6 +126,7 @@ sub loadtest ($script, %args) {
             sys.path.append("$inc")
             sys.path.append("$script_dir")
             import $name
+            print("Using python version: %s" % sys.version)
```
Comment (okurz) [inline autotest.pm:129]: IMHO we can use better approaches than an unconditional print. How about using python logging and a log.debug here? I don't see enough value to make that an unconditional print.

**Review by okurz — CHANGES_REQUESTED** [review]

File: autotest.pm (hunk 2 above)
Comment (josegomezr) [reply autotest.pm:129]: `log.X` will still mean an unconditional print, python logger by default (and by no other interaction) has the lowest reporting level.

To be able to "skip" that message you'd be willing to do the undoable: alter the global defaults to skip a message print.

Comment (Martchus) [inline autotest.pm:129]: This would print the Python version before each module. It would likely be better to print it just once. And why is this needed anyways?

**Comment by Martchus** [comment]
I guess the title/commit message should use "its" and not "it's".

File: t/08-autotest.t (hunk 1 above)
Comment (foursixnine) [reply t/08-autotest.t:325]: @josegomezr that was my initial idea, for some reason this isn't captured properly

File: autotest.pm (hunk 2 above)
Comment (foursixnine) [reply autotest.pm]: @okurz if you suggest a different place to put the Python version, I'm cool with that, I honestly don't care about the unconditional print, with an unconditional print, we'd would clearly mark that X module is running with Python (see my answer to @Martchus )

Also this has to be perl.diag, so it shows up in the autoinst-log.txt

File: t/08-autotest.t (hunk 1 above)
Comment (josegomezr) [reply t/08-autotest.t:325]: I'm no regex expert, I understand the regex match you're using is:

`#(?^:(?^:scheduling pythontest tests/pythontest)+(?^:Using python version))#`

against:

```
[2023-10-17T09:26:39.922954Z] [debug] [pid:2061] scheduling pythontest tests/pythontest.py
```

The first part of the regex  `$python_module_regex` matches, but the second it's just immediately after, it's expecting `$python_version_regex` (or `Using python version`) to be immediately after the module name.

I think you're printing two lines instead. See this regex101:

https://regex101.com/r/jsB2gf/1

File: t/08-autotest.t [hunk 3]
```diff
@@ -316,9 +316,13 @@ subtest 'load test successfully when CASEDIR is a relative path' => sub {
 };
 
 subtest python => sub {
-    stderr_like {
+    my $python_module_regex = qr{scheduling pythontest tests/pythontest};
+    my $python_version_regex = qr{Using python version};
+
```
Comment (foursixnine) [inline t/08-autotest.t:321]:
```suggestion
```

File: t/08-autotest.t [hunk 4]
```diff
@@ -316,9 +316,10 @@ subtest 'load test successfully when CASEDIR is a relative path' => sub {
 };
 
 subtest python => sub {
-    stderr_like {
+    combined_like {
         lives_ok { autotest::loadtest('tests/pythontest.py') } 'can load test module'
-    } qr{scheduling pythontest tests/pythontest.py}, 'python pythontest module referenced';
+    } qr{(?|scheduling pythontest tests/pythontest|Using python version)}, 'python pythontest module referenced';
```
Comment (perlpunk) [inline t/08-autotest.t:321]: That regex would even match an empty string.
I think what you would want here is:
```suggestion
    } qr{scheduling pythontest tests/pythontest.*Using python version}s, 'python pythontest module referenced';
```

**Comment by perlpunk** [comment]
I'll try to prepare a PR which prints the python version only when loading a python test for the first time
